swagger_server/controllers/distance_controller.py

```diff
@@ -45,16 +49,21 @@ def samples_id_nearest_neighbours_get(id):  # noqa: E501
     """
 
     try:
-        node = SampleNode.nodes.get(name=id)
-        neighbors = node.neighbors.all()
-        rels = [node.neighbors.relationship(n) for n in neighbors]
+        result = db.Database.get().query(
```

Comment:
I still don't understand why this has to be different from the one used in the `nearest-leaf-node` relationship query. In both cases, if either the sample node itself or the relationships do not exist, the response should be always `404`, because that means the resource that is being requested is `NOT FOUND`.

Reply:
Well, I thought that since we constrained every node to have a lineage, it's impossible for a node to not have one, so we don't use `OPTIONAL MATCH` (for the lineage)
But for nearest neighbours, some nodes can have no neighbours, therefore it makes sense to use `OPTIONAL MATCH` (for the neighbours)
But if you want the API to return 404 in case of no neighbour that's ok too

Reply:
The API does not require there is always one nearest leaf associated with a sample so we can't make this constraint at database level. So there are chances in which a sample could have no nearest leaf.

The API (`/samples/{id}/nearest-neighbours`) should return 404 if nearest neighbours are not found, because the API is requesting a resource which is not there.